thoth/adviser/product.py

```diff
@@ -127,10 +130,19 @@ def from_final_state(cls, *, context: Context, state: State) -> "Product":
             runtime_environment=context.project.runtime_environment,
         )
 
+        justification_metadata: List[Dict[str, Any]] = []
+        metadata = os.getenv("THOTH_ADVISER_METADATA")
+        if metadata:
+            try:
+                metadata_content = json.loads(metadata)
+                justification_metadata = (metadata_content.get("thoth.adviser") or {}).get("justification") or []
+            except Exception:
+                _LOGGER.exception("Failed to parse adviser metadata")
+
         return cls(
             project=advised_project,
             score=state.score,
-            justification=state.justification,
+            justification=list(chain(justification_metadata, state.justification)),
```

Comment:
what about `justification_metadata + (state.justification or [])`

Reply:
Adjusted in the most recent commit :+1: